amd_openvx_extensions/amd_rpp/source/Blend.cpp

```diff
@@ -0,0 +1,259 @@
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct BlendLocalData {
+    RPPCommonHandle * handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pSrc_2;
+    RppPtr_t pDst;
+    vx_float32 *shfit;
+    RpptDescPtr srcDescPtr;
+    RpptDesc srcDesc;
+    RpptDesc dstDesc;
+    RpptDescPtr dstDescPtr;
+    void *roiTensorPtr;
+    RpptROI *roiPtr;
+    RpptRoiType roiType;
+    Rpp32s inputLayout;
+    Rpp32s outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t ouputTensorDims[RPP_MAX_TENSOR_DIMS];
+    vx_enum inputTensorType;
+    vx_enum outputTensorType;
+};
+
+static vx_status VX_CALLBACK refreshBlend(vx_node node, const vx_reference *parameters, vx_uint32 num, BlendLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->srcDescPtr->n, sizeof(vx_float32), data->shfit, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_HIP
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HIP, &data->roiTensorPtr, sizeof(data->roiTensorPtr)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HIP, &data->pSrc, sizeof(data->pSrc)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HIP, &data->pSrc_2, sizeof(data->pSrc_2)));
+
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HIP, &data->pDst, sizeof(data->pDst)));
+#endif
+    } else if (data->deviceType == AGO_TARGET_AFFINITY_CPU) {
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HOST, &data->roiTensorPtr, sizeof(data->roiTensorPtr)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HOST, &data->pSrc, sizeof(data->pSrc)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HOST, &data->pSrc_2, sizeof(data->pSrc_2)));
+
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HOST, &data->pDst, sizeof(data->pDst)));
+    }
+    data->roiPtr = (RpptROI *)data->roiTensorPtr;
+    if((data->inputLayout == 2 || data->inputLayout == 3)) { // For NFCHW and NFHWC formats
+        unsigned num_of_frames = data->inputTensorDims[1]; // Num of frames 'F'
+        for(int n = data->srcDescPtr->n - 1; n >= 0; n--) {
+            unsigned index = n * num_of_frames;
+            for(int f = 0; f < num_of_frames; f++) {
+                data->shfit[index + f] = data->shfit[n];
+                data->roiPtr[index + f].xywhROI.xy.x = data->roiPtr[n].xywhROI.xy.x;
+                data->roiPtr[index + f].xywhROI.xy.y = data->roiPtr[n].xywhROI.xy.y;
+                data->roiPtr[index + f].xywhROI.roiWidth = data->roiPtr[n].xywhROI.roiWidth;
+                data->roiPtr[index + f].xywhROI.roiHeight = data->roiPtr[n].xywhROI.roiHeight;
```

Comment:
```suggestion
                data->roiPtr[index + f].xywhROI.xy.x = data->roiPtr[n].xywhROI.xy.x;
                data->roiPtr[index + f].xywhROI.xy.y = data->roiPtr[n].xywhROI.xy.y;
                data->roiPtr[index + f].xywhROI.roiWidth = data->roiPtr[n].xywhROI.roiWidth;
                data->roiPtr[index + f].xywhROI.roiHeight = data->roiPtr[n].xywhROI.roiHeight;
```

Change these to `data->roiPtr[index + f].xywhROI = data->roiPtr[n].xywhROI`